File: server/src/app/board/schemas.py
from datetime import datetime

from pydantic import BaseModel


class BoardCardCreate(BaseModel):
    team_uuid: str
    sprint_uuid: str
    title: str
    column_key: str = "todo"
    description: str | None = None
    position: int = 0
    assignee_user_uuid: str | None = None
    reporter_user_uuid: str | None = None


class BoardCardRead(BaseModel):
    uuid: str
    team_uuid: str
    sprint_uuid: str
    title: str
    column_key: str
    description: str | None
    position: int
    assignee_user_uuid: str | None
    reporter_user_uuid: str | None
    created_at: datetime
    updated_at: datetime


class BoardCardUpdate(BaseModel):
    title: str | None = None
    column_key: str | None = None
    description: str | None = None
    position: int | None = None
    assignee_user_uuid: str | None = None
    reporter_user_uuid: str | None = None


class BoardCardMove(BaseModel):
    """Move a card to a different column and/or position."""

    column_key: str
    position: int


class BoardCardPositionItem(BaseModel):
    uuid: str
    column_key: str
    position: int


class BoardCardBulkPositionUpdate(BaseModel):
    """Reorder multiple cards at once (after drag-and-drop)."""

    items: list[BoardCardPositionItem]

File: apps/web/src/entities/board/api/schemas.ts
import { z } from 'zod'

export const BoardCardReadSchema = z.object({
  uuid: z.string(),
  teamUuid: z.string(),
  sprintUuid: z.string(),
  title: z.string(),
  columnKey: z.string(),
  description: z.string().nullable(),
  position: z.number(),
  assigneeUserUuid: z.string().nullable(),
  reporterUserUuid: z.string().nullable(),
  createdAt: z.string(),
  updatedAt: z.string(),
})
export type BoardCardRead = z.infer<typeof BoardCardReadSchema>

export interface BoardCardCreate {
  readonly teamUuid: string
  readonly sprintUuid: string
  readonly title: string
  readonly columnKey?: string
  readonly description?: string | null
  readonly position?: number
  readonly assigneeUserUuid?: string | null
  readonly reporterUserUuid?: string | null
}

export interface BoardCardUpdate {
  readonly title?: string
  readonly columnKey?: string
  readonly description?: string | null
  readonly position?: number
  readonly assigneeUserUuid?: string | null
  readonly reporterUserUuid?: string | null
}

export interface BoardCardMove {
  readonly columnKey: string
  readonly position: number
}
